apps/server/src/modules/authorization/domain/rules/board-do.rule.ts

```diff
@@ -16,30 +23,84 @@ export class BoardDoRule implements Rule {
 
 	public hasPermission(user: User, boardDoAuthorizable: BoardDoAuthorizable, context: AuthorizationContext): boolean {
 		const hasPermission = this.authorizationHelper.hasAllPermissions(user, context.requiredPermissions);
-		if (hasPermission === false) {
+		if (!hasPermission) {
 			return false;
 		}
 
-		const userBoardRole = boardDoAuthorizable.users.find(({ userId }) => userId === user.id);
-		if (!userBoardRole) {
+		const userBoardRoles = boardDoAuthorizable.users.find(({ userId }) => userId === user.id);
+		if (!userBoardRoles) {
 			return false;
 		}
 
-		if (boardDoAuthorizable.requiredUserRole && boardDoAuthorizable.requiredUserRole !== userBoardRole.userRoleEnum) {
+		// permission< for elements under a submission item are handled by the parent submission item
+		if (isSubmissionItem(boardDoAuthorizable.parentDo)) {
+			const parentDoAuthorizable = {
+				boardDo: boardDoAuthorizable.parentDo,
+				users: boardDoAuthorizable.users,
+			} as BoardDoAuthorizable; // TODO fix this
+
+			if (boardDoAuthorizable.boardDo && isSubmissionItemContent(boardDoAuthorizable.boardDo)) {
+				return this.hasPermission(user, parentDoAuthorizable, context);
+			}
+			return false;
+		}
+
+		// files from drawingElement
+		// no matter the action, as long as the user has read permission on the drawing element, he has write permission on the file
+		if (isDrawingElement(boardDoAuthorizable.parentDo)) {
+			const parentDoAuthorizable = boardDoAuthorizable;
+			parentDoAuthorizable.boardDo = boardDoAuthorizable.parentDo;
+			parentDoAuthorizable.parentDo = undefined;
+			const drawElementContext = { ...context, action: Action.read };
+			return this.hasPermission(user, parentDoAuthorizable, drawElementContext);
+		}
+
+		if (context.action === Action.write) {
+			return isSubmissionItem(boardDoAuthorizable.boardDo)
+				? this.hasSubmissionItemWritePermission(userBoardRoles, boardDoAuthorizable.boardDo)
+				: this.isEditor(userBoardRoles);
+		}
+
+		return isSubmissionItem(boardDoAuthorizable.boardDo)
+			? this.hasSubmissiontemReadPermission(userBoardRoles, boardDoAuthorizable.boardDo)
+			: this.isReader(userBoardRoles);
```

Comment:
This function is a bit to complicated.

Try to have a top level catch.

if( submission case) return handleSubmissionCase(...);
if ( drawing case ) return handleDrawingCase(...);

if (context.action === Action.write) return this.isEditor;
if (context.action === Action.read) return this.isReader;

return false;

// maybe a switch statement is better